exp/addons/internal/controllers/clusterresourcesetbinding_controller.go

```diff
@@ -77,8 +78,33 @@ func (r *ClusterResourceSetBindingReconciler) Reconcile(ctx context.Context, req
 		// Error reading the object - requeue the request.
 		return ctrl.Result{}, err
 	}
-
-	cluster, err := util.GetOwnerCluster(ctx, r.Client, binding.ObjectMeta)
+	// Before 1.4 cluster name was stored as ownerRef; this code migrated this value to
+	// the spec.clusterName field if necessary.
+	clusterName := binding.Spec.ClusterName
+	if clusterName == "" {
+		// Initialize the patch helper.
+		patchHelper, err := patch.NewHelper(binding, r.Client)
+		if err != nil {
+			log.Error(err, "Failed to configure the patch helper")
+			return ctrl.Result{}, err
+		}
+		// Update the clusterName field of the existing ClusterResourceSetBindings with ownerReferences.
+		// More details please refer to: https://github.com/kubernetes-sigs/cluster-api/issues/7669.
+		if clusterName, err = getClusterNameFromOwnerRef(binding.ObjectMeta); err != nil {
+			return ctrl.Result{}, err
+		} else if clusterName == "" {
+			log.Info("ownerRef not found for the ClusterResourceSetBinding")
+			return ctrl.Result{}, nil
+		}
+		binding.Spec.ClusterName = clusterName
```

Comment:
q: should we drop the cluster owner ref? 
cc @killianmuldoon to also evaluate impacts on the periodic E2E test

Reply:
> should we drop the cluster owner ref?

Hi, @killianmuldoon what do you think?

Reply:
Yeah - good idea to drop it. We can update the e2e test as a follow-up IMO. FYI - Tuesday 14th March 2023 is the code freeze date. 

@chaunceyjiang do you think you can rebase this? If you can make the above change dropping the cluster ownerReference + rebase I'll review ASAP

Reply:
@killianmuldoon If we drop the ownerRef here and adjus the e2e test in a follow-up, CI will be broken on main, right?

I would like to avoid this if possible (especially so close to the release)

Reply:
> I would like to avoid this if possible (especially so close to the release)

+1

Reply:
Also fine to fix it here - but it would be periodics-only and a quick-fix. Just trying to help to move this RP along.

Reply:
Only breaking periodics and e2e-full on PRs is still pretty bad :/ 

Isn't the change to the e2e test relatively simple?

Reply:
I hoped it is just dropping "clusterGVK" in this line: https://github.com/kubernetes-sigs/cluster-api/blob/dd50972c6a4375a30265dc09ee1526556c3ea50b/test/framework/ownerreference_helpers.go#L162

Reply:
>Only breaking periodics and e2e-full on PRs is still pretty bad :/

I was going to follow up directly, just didn't want to complicate this PR to give it a better chance of getting in ahead of the deadline. As I said, fine to fix it here.

Reply:
So, we need to remove the cluster owner ref, right?

Reply:
Yes. Otherwise `pull-cluster-api-e2e-full-main` breaks.

If you can drop it from the ownerreference_helpers.go we can trigger the e2e-full test to see if it works (which it should)

Reply:
Done.